stdlib/format.c

```diff
@@ -239,3 +239,33 @@ uint256dec(char *output, uint256_t *val256)
 
     return output;
 }
+
+static const char b58digits[] = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+
+// https://github.com/bitcoin/libbase58/blob/b1dd03fa8d1be4be076bb6152325c6b5cf64f678/base58.c inspired this code.
+void base58_encode(uint8_t * data, uint32_t data_len, uint8_t * output, uint32_t output_len)
+{
+    uint32_t j, carry, zero_count = 0;
+
+    while (zero_count < data_len && !data[zero_count])
+        ++zero_count;
+
+    for (uint32_t i=zero_count, high = output_len - 1; i < data_len; i++, high = j)
+    {
+        for (carry = data[i], j = output_len - 1; (j > high) || carry; --j)
+        {
+            carry += 256 * output[j];
+            output[j] = carry % 58;
+            carry /= 58;
+            if (!j)
+            {
+                break;
+            }
+        }
+    }
+
```

Comment:
Why does the bitcoin implementation memset the leading null bytes to 1 while we don't?

Reply:
That's the way solana addresses are encoded. For example the systems account is all zeros (in hex/binary) and represented as `11111111111111111111111111111111` and not as a empty string

Reply:
Fixed